src/main/java/org/folio/rest/impl/InvoiceHelper.java

```diff
@@ -191,6 +193,10 @@
 
   private void setSystemGeneratedData(Invoice invoiceFromStorage, Invoice invoice) {
     invoice.withFolioInvoiceNo(invoiceFromStorage.getFolioInvoiceNo());
+    if (isTransitionToApproved(invoiceFromStorage, invoice)) {
```

Comment:
It would have been more obvious to me that it was part of the `InvoiceHelper#approveInvoice()` method.